### PR TITLE
Cardinal: 22.11 -> 22.12

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/cardinal/default.nix [hunk 1]
```diff
@@ -21,22 +25,26 @@
...
   prePatch = ''
     patchShebangs ./dpf/utils/generate-ttl.sh
   '';
 
+  dontUseCmakeConfigure = true;
   enableParallelBuilding = true;
 
   nativeBuildInputs = [ pkg-config ];
   buildInputs = [
+    cmake
```
Comment (symphorien) [inline pkgs/applications/audio/cardinal/default.nix:45]: cmake and file are presumably nativeBuildInputs. set `strictDeps` to true if unsure.

Comment (magnetophon) [reply pkgs/applications/audio/cardinal/default.nix:45]: Thanks!

File: pkgs/applications/audio/cardinal/default.nix [hunk 2]
```diff
@@ -48,6 +59,7 @@ stdenv.mkDerivation rec {
     libarchive
     liblo
     libsamplerate
+    libsndfile
     mesa
     python3
```
Comment (symphorien) [inline pkgs/applications/audio/cardinal/default.nix:64]: python3 is nativeBuildInput as well: the build fails for me with 
```
cardinal> /nix/store/561wgc73s0x1250hrgp7jm22hhv7yfln-bash-5.2-p15/bin/bash: line 1: python3: command not found
```

Comment (magnetophon) [reply pkgs/applications/audio/cardinal/default.nix]: I figured mesa is one as well.

Comment (symphorien) [reply pkgs/applications/audio/cardinal/default.nix]: mesa is a library, so that sounds suspicious. Maybe try building without mesa altogether...

**Comment by symphorien** [comment]
```
Cardinal --help
Failed to load JACK DLL, reason:
libjack.so.0: cannot open shared object file: No such file or directory
Failed to create the JACK client, reason was:
Communication error with the JACK server.
```
looks like there is a missing dependency on jack

**Comment by magnetophon** [comment]
When I have jack running, I get the same error.
Oddly enough, it does work with pipewire-jack.
Also strange: for that to work, it doesn't need libjack2 in the dependencies.

I did try building with jack in the deps, but got the same result.

Any ideas on how to troubleshoot further?
I tried doing a debug build, but it failed with: ``SUMMARY: AddressSanitizer: 9289095 byte(s) leaked in 5011 allocation(s).``

**Comment by magnetophon** [comment]
I got the standalone (jack) working.
I didn't manage to get the native (alsa/pulse) to work though, so I just removed it for now.
I doubt anyone will miss it, since Cardinal is a plugin first and foremost, the jack is an extra, and the alsa version is more of an afterthought.

I left in some dead code, so you can see what I tried and possibly make it work.
If noone manages, I'll just remove that code.

File: pkgs/applications/audio/cardinal/default.nix [hunk 3]
```diff
@@ -11,51 +14,72 @@
...
   enableParallelBuilding = true;
+  strictDeps = true;
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [
+    cmake
+    file
+    pkg-config
+    makeWrapper
+    mesa
```
Comment (symphorien) [inline pkgs/applications/audio/cardinal/default.nix:52]: it builds without mesa, please remove.

**Comment by symphorien** [comment]
thank you!

**Pull request merged**